alea/include/alps/alea/complex_op.hpp

```diff
@@ -201,6 +201,29 @@ class complex_op
                           inv_t * x.imre(), inv_t * (x.imim() + s));
     }
 
+    friend bool isnan(complex_op x)
+    {
+        return std::any_of(&x.vals_[0][0], &x.vals_[2][0],
+                           static_cast<bool(*)(T)>(std::isnan));
+    }
+
+    friend bool isfinite(complex_op x)
+    {
+        return std::all_of(&x.vals_[0][0], &x.vals_[2][0],
+                           static_cast<bool(*)(T)>(std::isfinite));
+    }
+
+    friend bool isinf(complex_op x)
+    {
+        // An object cannot be inf and nan at the same time, for multi-
+        // component objects nan takes precedence
+        if (isnan(x))
+            return false;
+
+        return std::any_of(&x.vals_[0][0], &x.vals_[2][0],
```

Comment:
are you sure about the 2? Should it be a 1? (same everywhere else in this block)

Reply:
No it's 2 here (ranges are exclusive).